packages/r/rttr/xmake.lua

```diff
@@ -0,0 +1,81 @@
+package("rttr")
+    set_homepage("https://www.rttr.org")
+    set_description("rttr: An open source library, which adds reflection to C++.")
+    set_license("MIT")
+
+    if is_plat("macosx") then
+        add_extsources("brew::rttr")
+    end
+    
+    add_urls("https://www.rttr.org/releases/rttr-$(version)-src.tar.gz",
+             "https://github.com/rttrorg/rttr/releases/download/v$(version)/rttr-$(version)-src.tar.gz",
+             "https://github.com/rttrorg/rttr.git")
+
+    add_versions("0.9.6", "f62caee43016489320f8a69145c9208cddd72e451ea95618bc26a49a4cd6c990")
+    add_versions("0.9.5", "caa8d404840b0e156f869a947e475b09f7b602ab53c290271f40ce028c8d7d91")
+
+    add_deps("cmake", "zlib")
+
+    on_install(function (package)
+        local configs = {}
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DBUILD_EXAMPLES=OFF")
+        table.insert(configs, "-DBUILD_DOCUMENTATION=OFF")
+        table.insert(configs, "-DBUILD_UNIT_TESTS=OFF") -- rttr has problem building unit tests on macosx.
+        -- rttr use BUILD_RTTR_DYNAMIC and BUILD_STATIC options to control whether to build dynamic or static libraries.
+        table.insert(configs, "-DBUILD_RTTR_DYNAMIC=" .. (package:config("shared") and "ON" or "OFF"))
+        table.insert(configs, "-DBUILD_STATIC=" .. (package:config("shared") and "OFF" or "ON"))
+        import("package.tools.cmake").install(package, configs, {cxxflags = "-Wno-error=implicit-const-int-float-conversion"})
```

Comment:
only for gcc/clang 

if package:has_tool("cxx", "gcc", "gxx", "clang", "clangxx") then